database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java

```diff
@@ -194,4 +199,6 @@ public int getItemViewType(int position) {
      * @param position  The position in the list of the view being populated
      */
     abstract protected void populateViewHolder(VH viewHolder, T model, int position);
+    
+    abstract protected void onCancelled(DatabaseError databaseError);
```

Comment:
I'd prefer to not introduce an abstract method that people won't implement properly. That's how we got stuck with this TODO to begin with.

How about logging it as a warning in the adapters and documenting that devs can override if they want different behavior?